src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -190,6 +191,22 @@ private sealed class CachedState
                         minimumVSDefinedSDKVersion);
                 }
 
+                string dotnetExe = Path.Combine(dotnetRoot, Constants.DotNet + Constants.ExeSuffix);
+                if (File.Exists(dotnetExe))
```

Comment:
This exists check could lead to a silent failure. Is that the right behavior?

Reply:
not sure. @baronfel should we throw an exception if "dotnet.exe" file does not exist? or we just don't set `DOTNET_HOST_PATH`?
If we decide to throw should this be under ChangeWave? Not sure how to achieve that in sdk repo. Probably to have 2 versions of this resolver

Reply:
If we switch to [/EnvironmentProvider.cs](https://github.com/dotnet/sdk/blob/8bd19a2390a6bba4aa80d1ac3b6c5385527cc311/src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs#L52), the info about missed dotnet will be logged in the method

Reply:
I decided to add logging in case when `DOTNET_HOST_PATH` or `SdkResolverMSBuildTaskHostRuntimeVersion` could not be set. Feel free to unresolve this conversation, if you decide we should throw